Embrace Unity iOS Interface/SpanRepository.swift

```diff
@@ -0,0 +1,92 @@
+//  Essentially identical to RN SpanRepository.Swift
+
+import Foundation
+import OSLog
+import OpenTelemetryApi
+
+// Should not get hit under normal circumstances, add as a guard against misinstrumentation
+private let MAX_STORED_SPANS = 10000
+
+class SpanRepository {
+    private let activeSpansQueue = DispatchQueue(label: "io.embrace.Unity.activeSpans", 
+                                                 attributes: .concurrent)
+    private let completedSpansQueue = DispatchQueue(label: "io.embrace.Unity.completedSpans", 
+                                                    attributes: .concurrent)
+    private var activeSpans = [String: Span]()
+    private var completedSpans = [String: Span]()
+    private var log = OSLog(subsystem: "Embrace", category: "SpanRepository")
+    
+    init() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(onSessionEnded),
+            name: Notification.Name("embrace.session.will_end"),
```

Comment:
If you import `EmbraceIO` (or `EmbraceCore`) you can use:
```swift
name: Notification.Name.embraceSessionWillEnd
```
This will prevent any kind of problems if we end up changing the name of that notification